src/tz_macos.rs

```diff
@@ -1,5 +1,41 @@
+use core_foundation_sys::base::{CFRelease, CFTypeRef};
+use core_foundation_sys::string::{kCFStringEncodingUTF8, CFStringGetCStringPtr};
+use core_foundation_sys::timezone::{CFTimeZoneCopySystem, CFTimeZoneGetName};
+
 pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
-    Ok(core_foundation::timezone::CFTimeZone::system()
-        .name()
-        .to_string())
+    unsafe {
+        Dropping::new(CFTimeZoneCopySystem())
+            .and_then(|tz| Dropping::new(CFTimeZoneGetName(tz.0)))
+            .and_then(|name| {
+                let name = CFStringGetCStringPtr(name.0, kCFStringEncodingUTF8);
+                if name.is_null() {
+                    None
+                } else {
+                    Some(name)
+                }
+            })
+            .and_then(|name| std::ffi::CStr::from_ptr(name).to_str().ok())
+            .map(|name| name.to_owned())
```

Comment:
This bit makes me a bit scared with the `Dropping` type and the `CStr` being dropped while a borrowed `&str` is taken from `name`. Is the `.map(|name| name.to_owned())` a use after free?

To be sure, I'd probably restructure all of these combinators to use short circuit return to make sure the `Dropping` and `CStr` wrappers are dropped in the right order.

Reply:
Hm, I am not exactly sure. An implementation that's guaranteed to be sane should be:

```rust
pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
    unsafe {
        if let Some(tz) = Dropping::new(CFTimeZoneCopySystem()) {
            if let Some(name) = Dropping::new(CFTimeZoneGetName(tz.0)) {
                let name = CFStringGetCStringPtr(name.0, kCFStringEncodingUTF8);
                if !name.is_null() {
                    if let Ok(name) = std::ffi::CStr::from_ptr(name).to_str() {
                        return Ok(name.to_owned());
                    }
                }
            }
        }
    }
    Err(crate::GetTimezoneError::OsError)
}
```

Reply:
to be double sure I'd promote the `std::ffi::CStr::from_ptr(name)` to a named binding before the `if let` instead of a temporary in the `if let`.